core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -118,6 +119,11 @@ void HIPInternal::print_configuration(std::ostream &s) const {
       << ", APU or dGPU: " << gpu_type
       << ", Is Large Bar: " << hipProp.isLargeBar
       << ", Supports Managed Memory: " << hipProp.managedMemory
+      << ", Architecture capable of accessing system allocated memory: "
+      << gpu_arch_can_access_system_allocations()
+      << ", System allows accessing system allocated memory on GPU: "
+      << (xnack_boot_config_has_hmm_mirror() and xnack_environment_enabled() and
```

Comment:
Any reason why you didn't use `&&` instead of `and`? It's not wrong but it's strange looking.

Reply:
No, just the first thing that came to my mind. I can change if `&&` is preferred

Reply:
Yes, please. Change it.

Reply:
done in https://github.com/kokkos/kokkos/pull/7673/commits/891ab31815697215d504b6615718031db89ce68a